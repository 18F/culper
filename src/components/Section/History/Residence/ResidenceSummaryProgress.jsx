import React from 'react'

import { i18n } from '@config'

import SummaryProgress from '@components/Section/History/SummaryProgress'
import { totalYears, excludeGaps } from '@components/Section/History/helpers'

import { Svg } from '@components/Form'
import { ResidenceValidator } from '@validators'

const dateRangeList = (items) => {
  const dates = []

  for (const i of excludeGaps(items)) {
    if (!i.Item) { continue }

    if (new ResidenceValidator(i.Item).isValid()) {
      dates.push(i.Item.Dates)
    }
  }

  return dates
}

const ResidenceSummaryProgress = (props) => {
  const { Residence, Birthdate } = props

  let residenceDates = []
  if (Residence && Residence.List && Residence.List.items) {
    residenceDates = Residence.List.items
  }

  return (
    <SummaryProgress
      className="residence"
      List={dateRangeList.bind(this, residenceDates)}
      title={i18n.t('history.residence.summary.title')}
      unit={i18n.t('history.residence.summary.unit')}
      total={totalYears(Birthdate)}>
        <div className="summary-icon">
          <Svg
            src="/img/residence-house.svg"
            alt={i18n.t('history.residence.summary.svgAlt')} />
        </div>
      </SummaryProgress>
  )
}

export default ResidenceSummaryProgress
