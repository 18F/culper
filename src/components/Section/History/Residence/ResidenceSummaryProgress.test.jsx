import React from 'react'
import { shallow } from 'enzyme'

import ResidenceSummaryProgress from './ResidenceSummaryProgress'

describe('The ResidenceSummaryProgress component', () => {
  it('renders without errors', () => {
    const component = shallow(
      <ResidenceSummaryProgress />
    )

    expect(component.exists()).toBe(true)
    expect(component).toMatchSnapshot()
  })
})
