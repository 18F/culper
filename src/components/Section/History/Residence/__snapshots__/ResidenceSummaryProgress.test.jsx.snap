// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The ResidenceSummaryProgress component renders without errors 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <SummaryProgress
    List={[Function]}
    className="residence"
    title="Where you have lived"
    total={10}
    unit="Years covered"
  >
    <div
      className="summary-icon"
    >
      <Svg
        alt="Years covered for locations you have lived"
        src="/img/residence-house.svg"
      />
    </div>
  </SummaryProgress>,
  "nodes": Array [
    <SummaryProgress
      List={[Function]}
      className="residence"
      title="Where you have lived"
      total={10}
      unit="Years covered"
    >
      <div
        className="summary-icon"
      >
        <Svg
          alt="Years covered for locations you have lived"
          src="/img/residence-house.svg"
        />
      </div>
    </SummaryProgress>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <ResidenceSummaryProgress
        years={10}
      />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "Birthdate": undefined,
          "Residence": undefined,
          "years": 10,
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <SummaryProgress
          List={[Function]}
          className="residence"
          title="Where you have lived"
          total={10}
          unit="Years covered"
        >
          <div
            className="summary-icon"
          >
            <Svg
              alt="Years covered for locations you have lived"
              src="/img/residence-house.svg"
            />
          </div>
        </SummaryProgress>,
        "_debugID": 2,
        "_renderedOutput": <SummaryProgress
          List={[Function]}
          className="residence"
          title="Where you have lived"
          total={10}
          unit="Years covered"
        >
          <div
            className="summary-icon"
          >
            <Svg
              alt="Years covered for locations you have lived"
              src="/img/residence-house.svg"
            />
          </div>
        </SummaryProgress>,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <ResidenceSummaryProgress
    years={10}
  />,
}
`;
